### DISCUSSION TITLE
Flexible Gene View in Gosling

### DISCUSSION THREAD

**Discussion by BioinfoUNIBA** [gosling-lang/gosling.js#1059] [General]
Hi all,

thank you for your Gosling visualization tool. I maintain a database for RNA editing events, called REDIportal. Inside the website, I included a GeneView page in which users can observe individual gene loci and alternative spliced isoforms as well as editing events. When the page is resized, gene view panels are resized accordingly. Here you can see an example http://srv00.recas.ba.infn.it/cgi/atlas/render_geneview.py?query9=hg&query10=hg19&query2=KRT13 .
I'm wondering if such a feature is also implemented in Gosling. I would like to include Gosling in my website and replace the current Gene View enabling the inclusion of additional tracks. However, I found that Gosling tracks were not resizable. Is there a way to implement this behavior?
Many thanks in advance.

**Comment by sehilyi** [comment]
HI! Thanks for reaching out! It sounds like all the features you are describing are supported in Gosling. For resizing, you can turn on the `responsiveSize` option:

```ts
{
  responsiveSize: { width: true }, // ← make the width of Gosling visualization fit its parent DOM element
  // ... rest of the Gosling spec
}
```

For example, you can visit the following example on the Editor to see it in action:

https://gosling.js.org/?gist=sehilyi/ab36a8321e48cf696ec10d7d48ba4912

Try to change the browser size, and the gosling visualization will adjust to the browser size.

An important thing to consider is what file format you are using. The Editor example above uses [BEDDB](https://docs.higlass.io/data_preparation.html), but the same visualization can be created using a regular [gff3](http://gosling-lang.org/) or [bed](http://gosling-lang.org/) file or plain text files (e.g., [csv](http://gosling-lang.org/)) in Gosling.

**Reply by BioinfoUNIBA** [reply]
Many thanks. It works properly in the Gosling Editor. However, when I export your example from the editor, it doesn't work. I don't know why. I tried to change the responsiveSize option but nothing happened. I don't guess the browser is the reason because the editor is open by the same.
